ui/packages/catalog-ui-search/src/main/webapp/component/query-basic/query-basic.view.js

```diff
@@ -45,17 +48,53 @@ function getMatchTypeAttribute() {
     : 'datatype'
 }
 
-function isTypeLimiter(filter) {
-  let typesFound = {}
-  filter.filters.forEach(subfilter => {
-    typesFound[CQLUtils.getProperty(subfilter)] = true
+const getMatchTypes = memoize(async () => {
+  const matchTypeAttr = getMatchTypeAttribute()
```

Comment:
✏️ this isn't being reused outside the single query request, maybe just inline it?